### PR TITLE
Expose DoVi metadata and VideoRangeType

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Server**
https://github.com/jellyfin/jellyfin/pull/7964

**Changes**
- Expose DoVi metadata and VideoRangeType

Please add this to **"stable backport" and "backend"**, thanks!

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

File: src/components/itemMediaInfo/itemMediaInfo.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ const attributeDelimiterHtml = layoutManager.tv ? '' : '<span class="hide">: </s
             if (stream.Profile) {
                 attributes.push(createAttribute(globalize.translate('MediaInfoProfile'), stream.Profile));
             }
-            if (stream.Level) {
+            if (stream.Level && stream.Type === 'Video') {
```
Comment (dmitrylyzo) [inline src/components/itemMediaInfo/itemMediaInfo.js:116]: Do we have to check `Type`?
Most attributes don't have such a check.

File: src/components/playerstats/playerstats.js [hunk 2]
```diff
@@ -269,31 +269,10 @@ import ServerConnections from '../ServerConnections';
...
-        }
-
-        if (videoStream.ColorTransfer) {
-            sessionStats.push({
-                label: globalize.translate('LabelColorTransfer'),
-                value: videoStream.ColorTransfer
-            });
-        }
-
-        if (videoStream.ColorPrimaries) {
```
Comment (dmitrylyzo) [inline src/components/playerstats/playerstats.js:293]: Removed as unnecessary (for the user)?

File: src/components/itemMediaInfo/itemMediaInfo.js (hunk 1 above)
Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: Yes, only for certain attributes. Such as level should only be shown for video type but not for music and image.

File: src/components/playerstats/playerstats.js (hunk 2 above)
Comment (nyanmisaka) [reply src/components/playerstats/playerstats.js:293]: The stats list is too long and it's better to replace these technical infos with "HDR10/HLG/DOVI".

File: src/components/itemMediaInfo/itemMediaInfo.js (hunk 1 above)
Comment (dmitrylyzo) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: Do we have `Level`, `AverageFrameRate` and `RealFrameRate` for non-video and `BitRate` for non-video/audio?

Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: Nope. The first three are video only and bitrate is audio/video only.

Comment (dmitrylyzo) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: Actually, I got `Level` for audio, but it is `0`.
I mean, maybe remove these checks?

Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: AKAIK 0 and -99 are usually invalid levels, compared with 51(5.1 h264) or 180(6.0 hevc). Did we ever use level attribute for audio?
https://github.com/jellyfin/jellyfin/blob/50bc41d84d7f3f7ecdccbb6ed1885426f87c0d9a/tests/Jellyfin.MediaEncoding.Tests/Test%20Data/Probing/music_metadata.json

Comment (dmitrylyzo) [reply src/components/itemMediaInfo/itemMediaInfo.js:116]: I have video VP9 with `Level: -99`.
Subtitles and audio have `Level: 0`.
I think we can just leave `if (Level)` or check for validity `if (Level > 0)`.

Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js]: Done.

File: src/components/itemMediaInfo/itemMediaInfo.js [hunk 3]
```diff
@@ -128,7 +128,7 @@ const attributeDelimiterHtml = layoutManager.tv ? '' : '<span class="hide">: </s
                 }
                 attributes.push(createAttribute(globalize.translate('MediaInfoInterlaced'), (stream.IsInterlaced ? 'Yes' : 'No')));
             }
-            if (stream.AverageFrameRate || stream.RealFrameRate) {
+            if ((stream.AverageFrameRate || stream.RealFrameRate) && stream.Type === 'Video') {
```
Comment (dmitrylyzo) [inline src/components/itemMediaInfo/itemMediaInfo.js:131]: This check seems necessary - the embedded cover image has `Level: -99` and `Framerate: 90000`.

File: src/components/itemMediaInfo/itemMediaInfo.js [hunk 4]
```diff
@@ -137,7 +137,7 @@ const attributeDelimiterHtml = layoutManager.tv ? '' : '<span class="hide">: </s
             if (stream.Channels) {
                 attributes.push(createAttribute(globalize.translate('MediaInfoChannels'), `${stream.Channels} ch`));
             }
-            if (stream.BitRate && stream.Codec !== 'mjpeg') {
+            if (stream.BitRate && (stream.Type === 'Video' || stream.Type === 'Audio')) {
```
Comment (dmitrylyzo) [inline src/components/itemMediaInfo/itemMediaInfo.js:140]: `BitRate: undefined` for the embedded cover image (`mjpeg`), so I think even the old check is unnecessary.

File: src/components/itemMediaInfo/itemMediaInfo.js (hunk 3 above)
Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js:131]: Removed.

File: src/components/itemMediaInfo/itemMediaInfo.js (hunk 4 above)
Comment (nyanmisaka) [reply src/components/itemMediaInfo/itemMediaInfo.js]: Removed.

**Review by dmitrylyzo — APPROVED** [review]

**Pull request merged**